lib/matplotlib/tests/test_image.py

```diff
@@ -1593,3 +1593,39 @@ def test_resample_dtypes(dtype, ndim):
     axes_image = ax.imshow(data)
     # Before fix the following raises ValueError for some dtypes.
     axes_image.make_image(None)[0]
+
+
+@pytest.mark.parametrize('intp_stage', ('data', 'rgba'))
+@check_figures_equal()
+def test_interpolation_stage_rgba_respects_alpha_param(fig_test, fig_ref, intp_stage):
+    axs_tst = fig_test.subplots(2, 3)
+    axs_ref = fig_ref.subplots(2, 3)
+    scl_a = 0.5  # scalar alpha
+    arr_a = np.random.rand(3, 3)  # array alpha
+
+    # When the image does not have an alpha channel, alpha should be specified
+    # by the user or default to 1.0
+    im1 = np.random.rand(3, 3, 3)
+    im_concat_arr_a = np.concatenate((im1, np.expand_dims(arr_a, axis=-1)), axis=-1)
```

Comment:
I believe this is more easy to read than concat/expand_dims:
```suggestion
    im_rgb = np.random.rand(ny, nx, 3)
    im_rgba = np.empty(ny, nx, 4)
    im_rgba[..., :3] = im_rgb
    im_rgba[..., 3] = array_alpha
```

Reply:
I prefer to avoid creating variables that are only used once.
How about this?
```
    im_rgb = np.random.rand(ny, nx, 3)
    im_concat_arr_a = np.concatenate(  # combine rgb channels with array alpha
        (im_rgb, array_alpha.reshape((3, 3, 1))), axis=-1
    )
```

Reply:
Sorry, my new commit made this outdated. Do you still prefer using temporary variables? If so I'd be happy to change it.

Reply:
No strong opinion. I find it slightly more readable to say "the first three components are `im_rgb`, the fourth component is `array_alpha` rather than "is a concantenation of `im_rgb` and `array_alpha` along the most inner axis.

Note that, you can write `array_alpha.reshape((3, 3, 1)` as `array_alpha[..., np.newaxis]`, which IMHO makes the intent a bit more clear. also `concatenate(..., axis=-1)` is equivalent to `dstack(...)` in this case. So
```
# combine rgb channels with array alpha
im_concat_arr_a = np.dstack((im_rgb, array_alpha[..., np.newaxis]))  
```
would also be an option.

I'll leave it to you to decide which way you want to go.